ambry-api/src/test/java/com.github.ambry/rest/MockBlobStorageService.java

```diff
@@ -313,6 +313,13 @@ public void onCompletion(BlobInfo result, Exception exception) {
       }
     } catch (Exception e) {
       exception = exception == null ? e : exception;
+      if (result != null && result.getBlobDataChannel() != null) {
```

Comment:
just pass the channel on - you probably don't need to do this - the `RequestResponseHandler` should close it eventually.